core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -143,9 +143,10 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
 
     protected val addressToExecutorId = new HashMap[RpcAddress, String]
 
-    // Spark configuration sent to executors. This is a lazy val so that subclasses of the
-    // scheduler can modify the SparkConf object before this view is created.
-    private lazy val sparkProperties = scheduler.sc.conf.getAll
+    // Spark configuration sent to executors. This is a method so that subclasses of the
+    // scheduler can modify the SparkConf object and latest spark configurations
+    // shall be sent to executors.
+    private def sparkProperties = scheduler.sc.conf.getAll
```

Comment:
Change `lazy val` to `def` to get the latest spark configurations. Will `lazy val`, new executor gets a snapshot of original spark conf and that will cause propagation of the initial log level to new executors.
To avoid that scenario, sparkProperties should not be a lazy val.

Reply:
This change has some performance disadvantages. What about storing the last `logLevel` received via `UpdateExecutorsLogLevel`?

Reply:
sure, I'll fix it without altering this `lazy val`

Reply:
You needed this `lazy val` => `def` change because you found a bug during the execution of the manual testing, right? In this case when you ready with the change please redo the testing.

Reply:
Update the code and manually  tested